addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -1227,6 +1227,12 @@ ListRenderer.include({
         var self = this;
         return this._selectRow(rowIndex).then(function () {
             var recordID = self._getRecordID(rowIndex);
+            if (!recordID) {
+                // This happens when clicking the last row of an editable=top list right
+                // after adding a new empty line. This causes the new line to be discarded
+                // and thus the row at the querried index doesn't exist anymore.
+                recordID = self._getBorderRow('last').data('id');
+            }
```

Comment:
So basically if I add a new row, and click on an existing one (not the last one) to discard it, it won't select the row I clicked on, but the next one?? If so, this it what should be fixed, imho, and it would fix the particular case of the last row as well.